ocaml/parsing/parsetree.mli

```diff
@@ -434,21 +434,19 @@ and expression_desc =
 and case =
     {
      pc_lhs: pattern;
-     pc_guard: guard option;
-     pc_rhs: expression;
+     pc_rhs: case_rhs;
    }
-(** Values of type {!case} represents [(P -> E)] or [(P when E0 -> E)] *)
-
-and guard =
-  | Guard_predicate of expression
-  | Guard_pattern of guard_pattern
-
-and guard_pattern =
-  {
-    pgp_scrutinee : expression;
-    pgp_pattern : pattern;
-    pgp_loc : Location.t;
-  }
+(** Values of type {!case} represents [(P -> E)], [(P when E0 -> E)],
+    [(P when E0 match P0 -> E)], or [(P when E match (cases))] *)
```

Comment:
Is it necessary to mention the third thing separately if it's a special case of the fourth?

Reply:
I also might prefer inlining the documentation as to what kind of case it is under the corresponding constructor of the variant.